lightning/src/ln/channel.rs

```diff
@@ -3842,14 +3842,8 @@ where
 			if ty != funding.get_channel_type() {
 				return Err(ChannelError::close("Channel Type in accept_channel didn't match the one sent in open_channel.".to_owned()));
 			}
-		} else if their_features.supports_channel_type() {
-			// Assume they've accepted the channel type as they said they understand it.
 		} else {
-			let channel_type = ChannelTypeFeatures::from_init(&their_features);
-			if channel_type != ChannelTypeFeatures::only_static_remote_key() {
-				return Err(ChannelError::close("Only static_remote_key is supported for non-negotiated channel types".to_owned()));
-			}
-			funding.channel_transaction_parameters.channel_type_features = channel_type;
+			return Err(ChannelError::close("channel_type assumed to be supported".to_owned()));
```

Comment:
Perhaps this is slightly better readable, accentuating the happy flow? Just a suggestion, in doubt whether it is better.

```rust
let ty = common_fields.channel_type.as_ref()
    .ok_or_else(|| ChannelError::close("channel_type assumed to be supported".to_owned()))?;

if ty != funding.get_channel_type() {
    return Err(ChannelError::close("Channel Type in accept_channel didn't match the one sent in open_channel.".to_owned()));
}
```